### PR TITLE
feat: create `@zk-kit/{eddsa,poseidon,semaphore}-artifacts` packages

### PR DESCRIPTION
**Pull request by sripwoud**
This PR:
- Creates artifacts packages
  - @zk-kit/eddsa-artifacts
  - @zk-kit/poseidon-artifacts
  - @zk-kit/semaphore-artifacts
- Defines a release GH action with [`changeset`](https://github.com/changesets/changesets) 

This PR releases/names the packages `@sripwoud/...` because I just want to test the release workflow once.
After this we can rename them correctly to @zk-kit and update the `NPM_TOKEN` secret.

Tested releasing manually `@sripwoud/eddsa-artifacts`: https://unpkg.com/@sripwoud/eddsa-artifacts@0.0.1/eddsa.wasm

### CODE REVIEW

File: packages/semaphore/package.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+	"name": "@sripwoud/semaphore-artifacts",
+	"description": "zk-kit semaphore artifacts",
+	"license": "AGPL-3.0",
```
Comment (cedoor) [inline packages/semaphore/package.json:4]: Why `AGPL-3.0`. Do you have a preference in this regard?

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+name: Release
+
+on:
+  push:
+    branches: ["main"]
+    paths:
```
Comment (cedoor) [inline .github/workflows/release.yaml:6]: How do `paths` work?

**Review by cedoor — APPROVED** [review]
LGTM 👍🏽  Just a few questions!

File: packages/semaphore/package.json (hunk 1 above)
Comment (sripwoud) [reply packages/semaphore/package.json:4]: I find AGPL healthier for OSS.
AGPL redefines what distribution means.
For GPL it just means "providing a copy" (download, CD, USB, etc...).
For AGPL it means providing network/online access to the software.
 It is very relevant given the "cloud" era.
image
Unfortunately most of the time GPL and MIT are still used by most proejcts.

Basically with AGPL anyone who accesses the software via a web service ("network distribution")built on it has the right to get the source code.
Not with GPL

Comment (cedoor) [reply packages/semaphore/package.json:4]: I see! Thanks

File: .github/workflows/release.yaml (hunk 2 above)
Comment (sripwoud) [reply .github/workflows/release.yaml:6]: The action will trigger only upon changes in the files listed in paths

Comment (cedoor) [reply .github/workflows/release.yaml:6]: Oh, thanks! I remember I started looking into it but then forgot. We should use them in the other repos as well

**Pull request merged**